### ISSUE TITLE
How close are we to generating PDF/UA compliant documents with HexaPDF?

### ISSUE THREAD

**Issue by freesteph** [gettalong/hexapdf#341]
Hello,

I've been using HexaPDF for a while and it's been a real treat so first and foremost thank you.

Since I work in gov we've got some accessibility requirements that we're keen to observe. I don't know much about PDF/UA or the PDF specs in general which is why I'm asking here instead of figuring it all out. I know it involves using a subset of PDF much like PDF/A which HexaPDF supports I believe. I've also gathered that tagging PDF is a must-have for UA, which I've seen planned somewhere in the last changelogs. 

Long story short, what's left to be able to claim PDF/UA compliant documents via HexaPDF ?

Thanks!

**Comment by gettalong** [comment]
Thanks!

As you said Tagged PDF is the most important part as it associates content with additional information to aid accessibility, context extraction, content reflowing...

Implementing support for Tagged PDF is one of three todos on my current todo list. This involves implementing the necessary PDF objects themselves, the convenience methods on top of them to make them easy to work with, and support for applying tagged PDF to content creation when using the document layout engine of HexaPDF.

So it should be easy to create a fully tagged PDF using HexaPDF. Adding the tagged PDF information to an existing PDF would also be possible but that would need to be implemented on a case-by-case basis by users.

**Comment by freesteph** [comment]
Thanks, looking forward to tagging support in HexaPDF then. Happy end of year! 🎄
